test/format/import.c

```diff
@@ -0,0 +1,223 @@
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "format.h"
+
+static void copy_file_into_directory(WT_SESSION *session, const char *dir, const char *name);
+static void import_with_repair(WT_SESSION *session);
+static void import_with_file_metadata(WT_SESSION *session, WT_SESSION *import_session);
+static void verify_import(WT_SESSION *session, int start_value);
+
+/*
+ * Import directory initialize command, remove and re-create the primary backup directory, plus a
+ * copy we maintain for recovery testing.
+ */
+#define HOME_IMPORT_INIT_CMD "rm -rf %s/IMPORT && mkdir %s/IMPORT"
+#define IMPORT_DIRNAME "IMPORT"
+#define IMPORT_URI "table:import"
+#define IMPORT_URI_CONFIG "key_format=i,value_format=i"
+#define IMPORT_ENTRIES 1000
+/*
+ * import --
+ *     Periodically import table.
+ */
+WT_THREAD_RET
+import(void *arg)
+{
+    WT_CONNECTION *conn, *import_conn;
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+    WT_SESSION *import_session, *session;
+    size_t len;
+    uint32_t import_value;
+    u_int period;
+    int counter;
+    char buf[2048];
+    char *cmd;
+
+    (void)(arg);
+    conn = g.wts_conn;
+    import_value = false;
+    counter = 0;
+
+    len = strlen(g.home) * 2 + strlen(HOME_IMPORT_INIT_CMD) + 1;
+    cmd = dmalloc(len);
+    testutil_check(__wt_snprintf(cmd, len, HOME_IMPORT_INIT_CMD, g.home, g.home));
+    testutil_checkfmt(system(cmd), "%s", "import directory creation failed");
+    free(cmd);
+
+    memset(buf, 0, sizeof(buf));
+    len = strlen(g.home) + strlen(IMPORT_DIRNAME) + 10;
+    cmd = dmalloc(len);
+    testutil_check(__wt_snprintf(cmd, len, "%s/%s", g.home, IMPORT_DIRNAME));
+    testutil_check(__wt_snprintf(
+      buf, sizeof(buf), "create,encryption=(name=%s)", encryptor(g.c_encryption_flag)));
+    /* Open a connection to the database, creating it if necessary. */
+    testutil_check(wiredtiger_open(cmd, NULL, buf, &import_conn));
+    free(cmd);
+
+    /* Open a session */
+    testutil_check(import_conn->open_session(import_conn, NULL, NULL, &import_session));
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+
+    testutil_checkfmt(
+      import_session->create(import_session, IMPORT_URI, IMPORT_URI_CONFIG), "%s", g.uri);
+
+    /*
+     * open_cursor can return EBUSY if concurrent with a metadata operation, retry in that case.
```

Comment:
This comment can be improved, also where is the retry logic?

Reply:
accidently forgot to remove.